doc/user/modes.md

```diff
@@ -0,0 +1,27 @@
+Fact Modes
+==========
+
+Fact modes only affect input. Compute semantics are the same regardless of the fact mode for the current attribute.
+
+The following fact modes are supported:
+
+1. Event
+2. Sparse States
+3. Dense States
+
+Event
+-----
+
+This is the default fact mode. Each fact is ingested as a single value.
+
+Sparse States
+-------------
+
+Facts are treated as state updates. A running state is kept and updated with each new fact.
+
+### Notes
+
+1. Number of facts: one extra fact with the current fact state is inserted with the snapshot/chord date.
```

Comment:
if it only kept the original date, it won't be counted in the window. So if the input only has:

```
bulbasaur|pokemon|{"location":"pallet"}|2014-01-01
```

and we ask for `feature pokemon ~> windowed  2 weeks ~> newest fields` at snapshot date `2016-1-1`, we will get nothing for bulbasaur.

On the other hand, if we add an extra fact at `2016-1-1`, we will get bulbasaur still being at pallet, which sounds right to me.